### PR TITLE
t2 run/push record an entry point for reuse by t2 restart. Fixes gh-675

### PR DESCRIPTION
**Pull request by rwaldron**
Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

**Comment by rwaldron** [comment]
In addition to closely monitoring appveyor, I will also reality check this on my windows machine and report back shortly

**Comment by rwaldron** [comment]
I've confirmed that Appveyor's "green" status is not a false positive: 



Locally, all tests pass on my windows 10 machine, however I cannot deploy due to some other unknown issue (nothing deploys, all I see is a "No Tessels Found" message)

**Comment by rwaldron** [comment]
cc @HipsterBrown @tikurahul or @johnnyman727

**Comment by HipsterBrown** [comment]
Looks good to me. Is there a smoke test I can run through?

**Comment by rwaldron** [comment]
@HipsterBrown sure: 


1. Complete `t2 init`
2. `t2 run index.js`, let it run then kill it
3. `t2 restart`


Expect: `t2 restart` should automatically restart from "index.js"

**Comment by tikurahul** [comment]
[LGTM](http://lgtm.in/i/Gt2iYurpn)

**Comment by rwaldron** [comment]
Fixes CR-0x4036d9b8

http://crash-reporter.tessel.io/crashes?fingerprint=0x4036d9b8

**Comment by HipsterBrown** [comment]
@rwaldron Ran through your smoke test with the following results:

```
nickhehr@Nicks-MacBook-Pro-2 tessel-test $ t2 run index.js
INFO Looking for your Tessel...
INFO Connected to tessel-router.
INFO Building project.
INFO Writing project to RAM on tessel-router (3.072 kB)...
INFO Deployed.
INFO Running index.js...
I'm blinking! (Press CTRL + C to stop)
^Cnickhehr@Nicks-MacBook-Pro-2 tessel-test $
nickhehr@Nicks-MacBook-Pro-2 tessel-test $ t2 restart
INFO Looking for your Tessel...
INFO Connected to tessel-router.
INFO Running undefined...
module.js:339
    throw err;
    ^

Error: Cannot find module '/tmp/remote-script/undefined'
    at Function.Module._resolveFilename (module.js:337:15)
    at Function.Module._load (module.js:287:25)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:134:18)
    at node.js:961:3
nickhehr@Nicks-MacBook-Pro-2 tessel-test $
```

I haven't started debugging yet but I'll check it out tonight.

**Comment by rwaldron** [comment]
Fixes CR-0xdc0fd98

http://crash-reporter.tessel.io/crashes?fingerprint=-0xdc0fd98

**Comment by rwaldron** [comment]
@HipsterBrown are you definitely using this branch?

**Comment by rwaldron** [comment]
Ah! I hit it too. Ok, will keep working on this.

**Comment by rwaldron** [comment]
@HipsterBrown ok, ready for review when it goes green

File: test/.jshintrc [hunk 1]
```diff
@@ -16,56 +16,57 @@
...
-    "Daemon": true,
-    "TesselSimulator": true,
-    "RemoteProcessSimulator": true,
-    "cli": true,
-    "LAN": true,
-    "TesselSeeker": true,
     "USB": true,
-    "tags": true,
-    "request": true,
+    "util": true,
```
Comment (rwaldron) [inline test/.jshintrc]: Sorry, this is just alphabetization 0_o

**Comment by HipsterBrown** [comment]
Did the smoke test again after the update, with the following results:

```
Nicks-MacBook-Pro:tessel-test hipsterbrown$ t2 init
Initializing Tessel 2 Project...
Created package.json.
Created .tesselinclude.
Wrote "Hello World" to index.js
Nicks-MacBook-Pro:tessel-test hipsterbrown$ t2 list
INFO Searching for nearby Tessels...
	USB	tessel-router
Nicks-MacBook-Pro:tessel-test hipsterbrown$ t2 run index.js
INFO Looking for your Tessel...
INFO Connected to tessel-router.
INFO Building project.
INFO Writing project to RAM on tessel-router (3.072 kB)...
INFO Deployed.
INFO Running index.js...
I'm blinking! (Press CTRL + C to stop)
^CNicks-MacBook-Pro:tessel-test hipsterbrown$ t2 restart
INFO Looking for your Tessel...
INFO Connected to tessel-router.
INFO Running index.js...
I'm blinking! (Press CTRL + C to stop)
^CNicks-MacBook-Pro:tessel-test hipsterbrown$
```

Success!!! :+1: LGTM

**Comment by rwaldron** [comment]
Thank you kindly sir ❤️

**Pull request merged**